### ISSUE TITLE
 pe.signatures[i].subject contains error

### ISSUE THREAD

**Issue by unmovedmoversqq** [VirusTotal/yara#1392]
[error.zip](https://github.com/VirusTotal/yara/files/5565040/error.zip)

I use digital signature matching：
pe.signatures[i].subject contains "Guangdong Fengqi Technology Co"
But there's no way to identify it.

**Comment by tlansec** [comment]
For some parts of the PE module you can see what's being parsed by YARA by using the -D flag when running your rule.

The signature is actually:

```
signatures
                [0]
                        thumbprint = "fd7c477549075beeda8b44971fc52ed807e186ee"
                        issuer = "/C=US/O=DigiCert Inc/OU=www.digicert.com/CN=DigiCert EV Code Signing CA"
                        subject = "/jurisdictionC=CN/jurisdictionST=\xE5\xB9\xBF\xE4\xB8\x9C\xE7\x9C\x81/jurisdictionL=\xE4\xB8\x9C\xE8\x8E\x9E\xE5\xB8\x82/businessCategory=Private Organization/serialNumber=91441900MA4X7P7J07/C=CN/ST=\xE5\xB9\xBF\xE4\xB8\x9C\xE7\x9C\x81"
                        version = 3
                        algorithm = "sha1WithRSAEncryption"
                        serial = "03:ba:dc:2b:62:ac:01:5b:d7:af:a7:76:18:0d:9f:a4"
                        not_before = 1603843200
                        not_after = 1626825599
```

I guess the other tool you're using is auto-translating this. You should be able to write a rule for the above though.

**Comment by wxsBSD** [comment]
Correct, YARA does not recognize the unicode characters used in the subject. You need to represent them as escaped bytes by just copy/pasting them from the output above (which you can get with the -D flag from yara itself).

**Comment by plusvic** [comment]
So I guess this can be closed.

**Comment by plusvic** [comment]
I think this is worth investigating. The subject extracted doesn't contain unicode characters, it contains the escaped characters literally. I mean, this don't match:

pe.signatures[0].subject contains "\xE5\xB9\xBF\xE4\xB8\x9C\xE7\x9C"

But this DOES match:

pe.signatures[0].subject contains "\\xE5\\xB9\\xBF\\xE4\\xB8\\x9C\\xE7\\x9C"

OpenSSL is returning the string already escaped, is this expected?
